Failed to compile hello-world example (AsyncRead is not implemented for std::io::Empty)

I tried to compile `examples/hello-world.rs` example on rust 1.39 beta and got an exception while compiling surf
```
error[E0277]: the trait bound `std::io::Empty: futures_io::if_std::AsyncRead` is not satisfied
  --> /home/akviatko/.cargo/registry/src/github.com-1ecc6299db9ec823/surf-1.0.2/src/http_client/mod.rs:64:21
   |
64 |             reader: Box::new(std::io::empty()),
   |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `futures_io::if_std::AsyncRead` is not implemented for `std::io::Empty`
   |
   = note: required for the cast to the object type `dyn futures_io::if_std::AsyncRead + std::marker::Send + std::marker::Unpin`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0277`.
error: could not compile `surf`.

To learn more, run the command again with --verbose.
```

* rust toolchain: beta-x86_64-unknown-linux-gnu
* rustc 1.39.0-beta.2 (5752b6348 2019-09-27)

I also created a [repo](https://github.com/flopacero/surf-helloworld) with example project that does not compile. It has `Cargo.lock` which might help to investigate the bug.

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -20,7 +20,7 @@ wasm-client = ["js-sys", "web-sys", "wasm-bindgen", "wasm-bindgen-futures"]
 middleware-logger = []
 
 [dependencies]
-futures-preview = { version = "0.3.0-alpha.18", features = ["compat", "io-compat"] }
+futures-preview = { version = "0.3.0-alpha.19", features = ["compat", "io-compat"] }
 http = "0.1.17"
 log = { version = "0.4.7", features = ["kv_unstable"] }
 mime = "0.3.13"

--- a/src/http_client/mod.rs
+++ b/src/http_client/mod.rs
@@ -61,7 +61,7 @@ impl Body {
     /// Create a new empty body.
     pub fn empty() -> Self {
         Self {
-            reader: Box::new(std::io::empty()),
+            reader: Box::new(futures::io::empty()),
         }
     }
 

